students/jack_anderson/lesson06/mailroom_pt4.py

```diff
@@ -0,0 +1,298 @@
+#!/usr/bin/env python3
+"""
+Jack Anderson
+02/24/2020
+UW PY210
+Lesson 06
+Mailroom assignment part 4
+"""
+from datetime import date
+import os
+
+# The list of donors and their amounts donated
+donors_list = [
+    ['Bubbles Trailer', [1500.99, 2523, 3012]],
+    ['Julien Park', [2520.92, 1623.12]],
+    ['Ricky Boys', [345.56, 5123.25]],
+    ['Jack Anderson', [1044, 2232, 4123.56]],
+    ['Lacey Coffin Greene', [1500.32, 625.34, 1305, 340.78]]
+  ]
+
+donors_dict = { donor[0]:donor[1] for donor in donors_list}
+
+
+def prompt_user(x):
+    """
+    Action to prompt user and handle keyboard interrupt
+    :param x: Prompt to display to user
+    :return: value input by user
+    """
+    try:
+        user_prompt = input(x)
+        if user_prompt.lower() == 'q':
+            start()
+        else:
+            return user_prompt
+    except(KeyboardInterrupt):
+        print()
+        quit()
+
+
+def prompt_name():
+    # Return the name entered by user
+    name = prompt_user("Enter the full name of the donor (enter 'q' to cancel or 'list' to view a list of donors): ")
+    namecheck = check_name(name)
+    return namecheck
+
+
+def prompt_amount():
+    # Return the donation amount entered by user
+    donation = prompt_user("Please enter a donation amount (enter 'q' to cancel): $ ")
+    donation_check = check_donation(donation)
+    return donation_check
+
+def list_names():
+    # Return a list of donors
+    donors = list(donors_dict)
+    return donors
+
+
+def add_items(name, donation):
+    """
+    Action to add items to the donor_dict
+    :param name: Donor name
+    :param donation: Donation made by donor
+    :return: Donor name, Donor Donation
+    """
+    donated = []
+    donated.append(float(donation))
+    donor = name.title()
+    donors = list_names()
+    for i in donors:
+        if name.lower() == i.lower():
+            donations = donors_dict[donor]
+            total = donated + donations
+            donors_dict[donor] = total
+
+        else:
+            donors_dict[donor] = donated
```

Comment:
this is incorrect use of your dictionary data structure

here is how your logic should look like:
```
if donors_dict.get(donor):
    # update donor data
else:
   # add donor data
```